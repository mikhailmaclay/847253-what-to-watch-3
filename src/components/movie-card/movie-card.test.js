import React from 'react';
import renderer from 'react-test-renderer';

import MovieCard from './movie-card';

const movie = {
  id: 1,
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  genre: `Fantasy`,
  releaseDate: `2018`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
};

it(`<MovieCard/> should render correctly`, () => {
  const {id, name, poster} = movie;

  const result = renderer.create(
      <MovieCard id={id} name={name} poster={poster} onMovieCardMouseOver={() => {}}/>
  ).toJSON();

  expect(result).toMatchSnapshot();
});
