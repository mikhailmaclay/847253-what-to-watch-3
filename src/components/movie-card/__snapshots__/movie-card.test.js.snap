// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard/> should render correctly 1`] = `
<a
  className="small-movie-card catalog__movie-card"
  href="/films/1"
  onClick={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald"
      height="175"
      src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <span
      className="small-movie-card__link"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </span>
  </h3>
</a>
`;
